django_saml/views.py

```diff
@@ -118,9 +118,10 @@ def saml_acs(request):
             request.session['samlNameIdNameQualifier'] = saml_auth.get_nameid_nq()
             request.session['samlNameIdSPNameQualifier'] = saml_auth.get_nameid_spnq()
             request.session['samlSessionIndex'] = saml_auth.get_session_index()
-            if 'RelayState' in req['post_data'] \
-                    and OneLogin_Saml2_Utils.get_self_url(req) != req['post_data']['RelayState']:
-                url = saml_auth.redirect_to(req['post_data']['RelayState'])
+            relay_state = req['post_data'].get('RelayState')
+            if relay_state \
```

Comment:
```suggestion
            if relay_state is not None \
```

Reply:
It's actually the source of the bug. If you want to be explicit, it should be:
`if relay_state is not None and relay_state != ''`

Because it comes in as an empty string. I've tried to explain that in the commit message.

Reply:
Ah I see, thank you for the clarification. I still would prefer being explicit, could you make that change?
I think the best option would be:
```suggestion
            if relay_state is not None and len(relay_state) > 0\
```
As long as `relay_state` is guaranteed to be a `str`.